src/storage/lock_manager/lock_wait_context.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Holds the state of a lock-waiting `AcquirePessimisticLock` request.
+//!
+//! When an `AcquirePessimisticLock` request meets a lock and enters
+//! lock-waiting state, it then may be either woken up by popping from the
+//! [`LockWaitingQueue`](super::lock_waiting_queue::LockWaitQueues),
+//! or cancelled by the
+//! [`WaiterManager`](crate::server::lock_manager::WaiterManager) due to
+//! timeout. [`LockWaitContext`] is therefore used to share the necessary state
+//! of a single `AcquirePessimisticLock` request, and ensuring the internal
+//! callback for returning response through RPC is called at most only once.
+//!
+//! Note: The corresponding implementation in `WaiterManager` is not yet
+//! implemented, and this mod is currently not used yet.
+
+use std::{
+    convert::TryInto,
+    result::Result,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+};
+
+use parking_lot::Mutex;
+use txn_types::TimeStamp;
+
+use crate::storage::{
+    errors::SharedError,
+    lock_manager::{lock_waiting_queue::PessimisticLockKeyCallback, LockManager, LockWaitToken},
+    Error as StorageError, PessimisticLockRes, ProcessResult, StorageCallback,
+};
+
+pub struct LockWaitContextInner {
+    #[allow(dead_code)]
+    pr: ProcessResult,
+    cb: StorageCallback,
```

Comment:
Could you add comments _here_ about what `pr` and `cb` actually is and when they will be called or used. It will be easier for readers to learn the code.